test/pkg/firewall/firewall.go

```diff
@@ -1,126 +1,335 @@
+	for i := 0; i < 2; i++ {
+		if len(parts1) > i && len(parts2) > i {
+			if parts1[i] > parts2[i] {
+				return true
+			} else if parts1[i] < parts2[i] {
+				return false
+			}
+		} else if len(parts1) > i {
+			// If v2 has no more parts, v1 is greater
+			return true
+		}
 	}
 
-	output, err := utilsHelpers.RunCommandOnPod(debugPod, []string{"bash", "-c", command})
+	// If they are equal or v1 is less than v2
+	return false
+}
+
+func createButaneConfig(nftablesRules, nodeRole string) []byte {
+	lines := strings.Split(nftablesRules, "\n")
+	nftablesRulesWithoutFirstLine := ""
+	if len(lines) > 1 {
+		nftablesRulesWithoutFirstLine = strings.Join(lines[1:], "\n")
+	}
+
+	butaneConfig := fmt.Sprintf(`variant: openshift
+version: 4.16.0
+metadata:
+  name: 98-nftables-cnf-%s
+  labels:
+    machineconfiguration.openshift.io/role: %s
+systemd:
+  units:
+    - name: "nftables.service"
+      enabled: true
+      contents: |
+        [Unit]
+        Description=Netfilter Tables
+        Documentation=man:nft(8)
+        Wants=network-pre.target
+        Before=network-pre.target
+        [Service]
+        Type=oneshot
+        ProtectSystem=full
+        ProtectHome=true
+        ExecStart=/sbin/nft -f /etc/sysconfig/nftables.conf
+        ExecReload=/sbin/nft -f /etc/sysconfig/nftables.conf
+        ExecStop=/sbin/nft 'add table inet openshift_filter; delete table inet openshift_filter'
+        RemainAfterExit=yes
+        [Install]
+        WantedBy=multi-user.target
+storage:
+  files:
+    - path: /etc/sysconfig/nftables.conf
+      mode: 0600
+      overwrite: true
+      contents:
+        inline: |
+          table inet openshift_filter
+          delete table inet openshift_filter
+		%s
+        `, nodeRole, nodeRole, nftablesRulesWithoutFirstLine)
+	butaneConfig = strings.ReplaceAll(butaneConfig, "\t", "  ")
+	return []byte(butaneConfig)
+}
+
+func fail(format string, args ...interface{}) {
+	fmt.Fprintf(os.Stderr, format, args...)
+	os.Exit(1)
+}
+
+func ConvertButaneToYAML(butaneContent []byte) ([]byte, error) {
+	options := common.TranslateBytesOptions{}
+
+	dataOut, _, err := butaneConfig.TranslateBytes(butaneContent, options)
 	if err != nil {
-		return nil, fmt.Errorf("failed to execute command '%s' on node %s: %w", command, debugPod.Spec.NodeName, err)
+		fail("failed to  : %v\n", err)
 	}
-	return output, nil
+
+	return dataOut, nil
 }
 
-// Checks if the include statement for the new nftables configuration,
-// Is present in the nftables.conf file. If the include statement is not present,
-// Add it the statement to the file.
-func editNftablesConf(debugPod *v1.Pod, utilsHelpers utils.UtilsInterface) error {
-	checkCommand := `grep 'include "/etc/nftables/firewall.nft"' /host/etc/sysconfig/nftables.conf | wc -l`
-	output, err := RunRootCommandOnPod(debugPod, checkCommand, false, utilsHelpers)
+func applyYAMLWithOC(output []byte, c *client.ClientSet, utilsHelpers utils.UtilsInterface, artifactsDir, role string) error {
```

Comment:
It's applying with oc, it's using client api
Can you use the butane api https://github.com/coreos/butane

Reply:
yah i used api not cmd